### ISSUE TITLE
Obok decrypted books and pages have random symbols

### ISSUE THREAD

**Issue by CheesyChucky** [apprenticeharper/DeDRM_tools#837]
Hey there.
I tried to decrypt Kobo Plus books but calibre convert them to epub but the books are all made out of random characters. 

> calibre 3.42  embedded-python: True is64bit: False
Windows-10-10.0.18898-SP0 Windows ('32bit', 'WindowsPE')
32bit process running on 64bit windows
('Windows', '10', '10.0.18898')
Python 2.7.15+
Windows: ('10', '10.0.18898', 'SP0', u'Multiprocessor Free')
Interface language: nl
Successfully initialized third party plugins: DeDRM (6, 6, 3) && Obok DeDRM (6, 5, 4)
Turning on automatic hidpi scaling
devicePixelRatio: 1.0
logicalDpi: 96.0 x 96.0
physicalDpi: 93.6046065259 x 93.6245733788
Using calibre Qt style: True
[0.00] Starting up...
[0.00] Showing splash screen...
[0.06] splash screen shown
[0.06] Initializing db...
[0.06] db initialized
[0.06] Constructing main UI...
DEBUG:    0.0 obok::utilities.py - loading translations
DEBUG:    0.0 obok::dialogs.py - loading translations
DEBUG:    0.0 obok::config.py - loading translations
DEBUG:    0.0 obok::action_err.py - loading translations
[1.00] main UI initialized...
[1.00] Hiding splash screen
[1.58] splash screen hidden
[1.58] Started up in 1.58 seconds with 0 books
Running Obok DeDRM v6.5.4
DEBUG:    4.4 didn't find device
Obok v3.2.4
Copyright © 2012-2016 Physisticated et al.
c:\users\censored\appdata\local\temp\tmpkwx08f
DEBUG:    4.4 got kobodir C:\Users\censored\AppData\Local\Kobo\Kobo Desktop Edition
Found 4 possible keys to try.
DEBUG:    7.4 BookListTableWidget:get_books - book_num: 0
DEBUG:    7.4 BookListTableWidget:get_books - book: De Toren van Babel staat in Brussel
DEBUG:    7.4 InterfacePluginAction::launchObok - number of books to decrypt: 1
Obok DeDRM v6.5.4 - Decrypting De Toren van Babel staat in Brussel
Trying key:  cc75fcee0243bea003d6c5beee32cd92
Success!
DEBUG:    7.6 DecryptAddProgressDialog::do_book_action - decrypted book: 'De Toren van Babel staat in Brussel'
Obok DeDRM v6.5.4 - Added De toren van babel staal in brussel
Obok DeDRM v6.5.4 - wrapping up results.
Worker Launch took: 0.207000017166

![Knipsel](https://user-images.githubusercontent.com/51167441/58567619-f4e2f100-8232-11e9-8b03-21f1af7cdc93.PNG)

**Comment by ElleKayEm** [comment]
Kobo Plus is a subscription service, right?  These tools are intended for purchased books.

**Comment by CheesyChucky** [comment]
> Kobo Plus is a subscription service, right? These tools are intended for purchased books.

Oh that make sense, i already had the psyhical copy of the book but wanted it digital, but now i know why it doesn't work.

**Comment by hholst80** [comment]
I have both a physical copy and I bought the book on Kobo thinking it was DRM free (that is a requirement for me). Is there any way I can get rid of the DRM and still have the output EPUB readable? Do I need to do anything with the xml file URLLink.acsm?

**Comment by ElleKayEm** [comment]
Obok should work on a purchased book from the Kobo Desktop program.  Or you can use DeDRM with the regular epub downloaded via Adobe Digital Editions 2.0.1 authorized with your Adobe ID.

**Comment by hholst80** [comment]
OK. I thought it would be possible to connect directly from Calibre + OBok to the Kobo E-ink reader without the need for the intermediate desktop software?

**Comment by ElleKayEm** [comment]
Oh, I think you can do that as well.  Not sure if there's anything you need to do.  Check the Obok plugin customization.

**Comment by lightbulbjim** [comment]
I just purchased two books and attempted to decrypt them using the Calibre plugin (directly from the ereader, I don't have Kobo Desktop installed). One worked and the other ended up as random symbols as reported in this issue. Reading the book on the ereader works fine.

This is the first time I've seen this problem. I'm not sure how to proceed.

**Comment by ElleKayEm** [comment]
Thanks for reporting what worked!  To confirm, you put your Kobo ereader's serial number into the Obok plugin customization?

**Comment by lightbulbjim** [comment]
> To confirm, you put your Kobo ereader's serial number into the Obok plugin customization?

Correct.

**Comment by Gerben321** [comment]
Did this work for the Kobo Plus epubs? Because out of 9 books that I wanted to import from Kobo Plus only one is readable. The obok importer says they are all DRM-free (that's what the Kobo site says as well). I already entered my SN from my Kobo, but no success.

**Comment by ElleKayEm** [comment]
Isn't Kobo Plus a subscription service?  A book that's _sold_ DRM-free still isn't going to be "leased" without DRM.

**Comment by Gerben321** [comment]
That's true I guess. Solved it by putting it on my Kobo first, and then importing it with Calibre. It then said the books actually _were_ protected by DRM. It all worked after importing. 👍

**Comment by syther101** [comment]
Seem to also be having this issue. Books which show as not being protected do not import correctly. These are all books purchased directly from Kobo and not Kobo Plus in my instance.

**Comment by rancilio2000** [comment]
I also have this issue for bought kobo books. Some of them works, some don't. 
it happens when dedrm from the kobo desktop.

**Comment by ElleKayEm** [comment]
@rancilio2000 Close Kobo Desktop before using Obok.  If it is a book with DRM, it should show a closed red lock in Obok.

eta: Also, update to Obok 10.0.9 if you aren't already.
